gridmenuview/src/main/java/com/bcc/gridmenuview/GridMenu.java

```diff
@@ -59,6 +72,25 @@ public void setOnClickListener(OnItemClickListener listener) {
         adapter.setOnClickListener(listener);
     }
 
+    private void setMenuItemShape() {
+        ShapeElement shapeLabel = ShapeElement.values()[shapeItem];
+        Shape shape;
+        switch (shapeLabel) {
+            case SQUARE:
+                shape = new Square(maskableFrameLayout, menuItemOverlay);
+                shape.maskShape();
+                shape.shapeOverlay();
+                break;
+            case CIRCLE:
+                shape = new Circle(maskableFrameLayout, menuItemOverlay);
+                shape.maskShape();
+                shape.shapeOverlay();
+                break;
+            default:
+                break;
```

Comment:
Flow to this branch is impossible right?
If so, I think it's better to just throw an exception. Let it crash instead of behaving unexpectedly or just default to the circle shape.
Also, move the line 86 and 87 to after the switch branching.